optlib/forth.ctags

```diff
@@ -0,0 +1,23 @@
+#
+#  Copyright (c) 2023, Eric Forgeot
+#  Copyright (c) 2023, Masatake YAMATO
+#
+#  This source code is released for free distribution under the terms of the
+#  GNU General Public License version 2 or (at your opinion) any later version.
+#
+#  This module contains functions for generating tags for Forth files
+#  https://www.forth.com/starting-forth/1-forth-stacks-dictionary/
+#
+--langdef=Forth
+--map-Forth=+.fth
+--map-Forth=+.forth
+# TODO: a selector arbitrating Fortran and Forth
```

Comment:
This is my job.

How can we distinguish a Fortran source file and a Forth source file?
I need heuristics.

Fortran:
- a line starts from `program
- a line starts from `!`

Forth:
- a line starts from `\`
- a line starts from `: `
- a line starts from a digit

What do you think?

Reply:
it should do the job. 
I don't know Fortran, but Forth gives much freedom for naming words, so it could also be legal to name a word ``!word``. ``!`` is even a command (for storing a value into a variable). As it's stack based, it would be possible to write:

```
50
SomeVariable
!
```

but most people would probably write instead ``50 SomeVariable !``

There are also many other commands which can start a line, but a file with a high number of ``:`` and ``\`` at the beginning of lines would probably be forth

Reply:
Thank you for the input. I will implement the selector.